activerecord/test/cases/tasks/database_tasks_test.rb

```diff
@@ -210,6 +210,27 @@ def test_establishes_connection_for_the_given_environment
         ActiveSupport::StringInquirer.new("development")
       )
     end
+
+    def test_creating_production_failure_does_not_dump_credentials
+      old_env = ENV["RAILS_ENV"]
+      ENV["RAILS_ENV"] = "production"
+
+      @configurations['production'].merge!('username'=>'users_uname',
+        'password'=>'users_password')
+      ActiveRecord::Tasks::DatabaseTasks.stubs(:class_for_adapter).raises(Exception)
+
+      $stderr.stubs(:puts).returns(true)
+      $stderr.expects(:puts).
+        with("Couldn't create database for #{@configurations["production"].except("username","password").inspect}")
```

Comment:
Instead of copying implementation code 

    @configurations["production"].except("username","password").inspect)

I will suggest to expects whole message or ensure actual password is not present in this string.